diff --git a/libavfilter/vsrc_amf.c b/libavfilter/vsrc_amf.c
index 2a9811ae4a..e1ea0dc0f7 100644
--- a/libavfilter/vsrc_amf.c
+++ b/libavfilter/vsrc_amf.c
@@ -74,7 +74,11 @@ AVFILTER_DEFINE_CLASS(amf_capture);
 // need to increase precision for capture timing accuracy
 #if defined (_WIN32)
 
+#if __has_include(<timeapi.h>)
 #include <timeapi.h>
+#else
+#include <mmsystem.h>
+#endif
 
 typedef MMRESULT (WINAPI *timeBeginPeriod_fn)(UINT uPeriod);
 typedef MMRESULT (WINAPI *timeEndPeriod_fn)(UINT uPeriod);
